system-addon/lib/TopSitesFeed.jsm

```diff
@@ -72,13 +72,24 @@ this.TopSitesFeed = class TopSitesFeed {
         this.getScreenshot(link.url);
       }
     }
-
+    return links;
+  }
+  async refresh(action) {
+    let links = await this.getLinks();
     const newAction = {type: at.TOP_SITES_UPDATED, data: links};
 
     // Send an update to content so the preloaded tab can get the updated content
     this.store.dispatch(ac.SendToContent(newAction, action.meta.fromTarget));
     this.lastUpdated = Date.now();
   }
+  async clearHistory(action) {
```

Comment:
This and `refresh` are doing almost the same thing, so unless we expect the logic to diverge much, this will likely cause problems if one is updated and not the other. Also the method name seems to imply it's trying to clear history instead of handling a cleared history.